File: rt-node-master/src/main/java/xyz/redtorch/node/master/web/socket/WebSocketServerConfig.java
package xyz.redtorch.node.master.web.socket;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.socket.config.annotation.EnableWebSocket;
import org.springframework.web.socket.config.annotation.WebSocketConfigurer;
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry;

@Configuration
@EnableWebSocket
public class WebSocketServerConfig implements WebSocketConfigurer {

	@Autowired
	private WebSockeServerHandshakeInterceptor webSockeServerHandshakeInterceptor;
	@Autowired
	private WebSocketServerHandler webSocketServerHandler;

	public void registerWebSocketHandlers(WebSocketHandlerRegistry registry) {
		registry.addHandler(webSocketServerHandler, "/websocket").setAllowedOrigins("*").addInterceptors(webSockeServerHandshakeInterceptor);
	}
}

File: rt-master/src/main/kotlin/xyz/redtorch/master/web/socket/WebSocketServerConfig.kt
package xyz.redtorch.master.web.socket

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Configuration
import org.springframework.web.socket.config.annotation.EnableWebSocket
import org.springframework.web.socket.config.annotation.WebSocketConfigurer
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry

@Configuration
@EnableWebSocket
open class WebSocketServerConfig : WebSocketConfigurer {

    @Autowired
    lateinit var webSocketServerHandshakeInterceptor: WebSocketServerHandshakeInterceptor

    @Autowired
    lateinit var tradeClientWebSocketHandler: TradeClientWebSocketHandler

    @Autowired
    lateinit var slaveNodeWebSocketHandler: SlaveNodeWebSocketHandler

    override fun registerWebSocketHandlers(registry: WebSocketHandlerRegistry) {
        registry.addHandler(tradeClientWebSocketHandler, "/ws/trade")//
            .addHandler(slaveNodeWebSocketHandler, "/ws/slave")//
            .setAllowedOrigins("*")//
            .addInterceptors(webSocketServerHandshakeInterceptor)//
    }
}